.github/workflows/java.yml

```diff
@@ -2,26 +2,98 @@ name: Java
 
 on:
   push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
 
 jobs:
-  test:
-    name: Test
+  build-and-test:
+    name: Build and Test
     strategy:
+      fail-fast: false
       matrix:
         os: [ ubuntu-latest ]
-        java: [ 8, 11, 17 ]
+        java: [ 8, 9, 10, 11, 12, 13, 14, 15, 16, 17 ]
```

Comment:
it would be nice to define a consistent approach between this library and the Java SDK. Currently we don't have a matrix on the library but we can onboard it

Reply:
I thought it could be useful. I just based this on other libraries' approaches such as [retrofit](https://github.com/square/retrofit/blob/master/.github/workflows/build.yml#L12)